core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelComponentUtils.java

```diff
@@ -199,102 +205,111 @@ public static boolean isUnsupportedProperty(Parameter p) {
-
-		return compClass;
+		return null;
 	}
-
+	
+	private static String getComponentClassFromJar(IJavaProject jpr, String scheme){
+		try {
+			for (IClasspathEntry e : jpr.getResolvedClasspath(true)) {
+				File cpEntryFile = e.getPath().toFile();
+				String compClass = getClassFromZipEntry(scheme, cpEntryFile);
+				if (!Strings.isBlank(compClass)) {
+					return compClass;
+				}
+			}
+		} catch (JavaModelException ex) {
+			CamelModelServiceCoreActivator.pluginLog().logError(ex);
+		}
+		return null;
+	}
+	
 	/**
 	 * returns the component class for the given scheme
 	 *
 	 * @param scheme
-	 * @param project
 	 * @return the class or null if not found
 	 */
-	protected static String getComponentJSon(String scheme, IProject project) {
-		String json = null;
-
-		try {
-			ZipFile zf = null;
-			ZipEntry ze = null;
+	protected static String getComponentClass(String scheme, CamelFile camelFile) {
+		String compClass = getComponentClassForScheme(scheme, camelFile);
+		if (compClass == null) {
+			// seems this scheme has no model entry -> check dependency
+			IProject project = camelFile.getResource().getProject();
 			IJavaProject jpr = JavaCore.create(project);
-			for (IClasspathEntry e : jpr.getResolvedClasspath(true)) {
-				File cpEntryFile = e.getPath().toFile();
-				if (isJarFile(cpEntryFile)) {
-					zf = new ZipFile(cpEntryFile);
-					ze = zf.getEntry(String.format("META-INF/services/org/apache/camel/component/%s", scheme));
-					if (ze != null) {
-						break;
-					}
-					ze = null;
-					zf = null;
-				}
-			}
+			compClass = getComponentClassFromJar(jpr, scheme);
+		}
 
+		return compClass;
+	}
+
+	private static String getComponentJsonFromJar(File cpEntryFile, String scheme) {
+		try (ZipFile zf = new ZipFile(cpEntryFile)) {
+			ZipEntry ze = zf.getEntry(String.format("META-INF/services/org/apache/camel/component/%s", scheme));
 			if (ze != null) {
 				// try to generate a model entry for the component class
 				Properties p = new Properties();
 				p.load(zf.getInputStream(ze));
 				String compClass = p.getProperty("class");
-				String packageName = compClass.substring(0, compClass.lastIndexOf("."));
+				String packageName = compClass.substring(0, compClass.lastIndexOf('.'));
 				String folder = packageName.replaceAll("\\.", "/");
 				ze = zf.getEntry(String.format("%s/%s.json", folder, scheme));
 				if (ze != null) {
-					json = IOUtils.loadText(zf.getInputStream(ze), null);
+					return IOUtils.loadText(zf.getInputStream(ze), null);
 				}
 			}
-		} catch (Exception ex) {
+		} catch (IOException ex) {
 			CamelModelServiceCoreActivator.pluginLog().logError(ex);
-			json = null;
 		}
-
-		return json;
+		return null;
+	}
+	
+	/**
+	 * returns the component class for the given scheme
+	 *
+	 * @param scheme
+	 * @param project
+	 * @return the class or null if not found
+	 */
+	protected static String getComponentJSon(String scheme, IProject project) {
+		IJavaProject jpr = JavaCore.create(project);
+		
+		try {
```

Comment:
What do you think to put the Try inside the loop?
like that if we have an invalid jars but which is not the one containing the JSON we are interested in, it will still work.